client/src/js/services/journalDataviewService.js

```diff
@@ -0,0 +1,205 @@
+        grid.render();
+      });      
+    };
+
+    dataviewService.subscribeToOnRowsChanged = function subscribeToOnRowsChanged (grid){
+      if(!grid){throw 'Undefined grid';}
+      dataviewService.dataview.onRowsChanged.subscribe(function (e, args) {
+        grid.invalidateRows(args.rows);
+        grid.render();
+      });
+    };
+
+    dataviewService.populate = function populate () {
+      return dataviewService.dataLoaderService.loadJournalRecord()
+      .then(function (data){
+        data.journalRecord.data = data.journalRecord.data.map(function (item) {
+          item.trans_date = new Date(item.trans_date);
+          return item;
+        });
+        dataviewService.dataview.beginUpdate();
+        dataviewService.dataview.setItems(data.journalRecord.data, 'uuid');
+        dataviewService.dataview.endUpdate();   
+      });      
+    };
+
+    dataviewService.addNewItem = function addNewItem(row){
+      dataviewService.dataview.addItem(row);
+    };
+
+    dataviewService.getItem = function getItem(row){
+      return dataviewService.dataview.getItem(row);
+    };
+
+    dataviewService.getItems = function getItem(){
+      return dataviewService.dataview.getItems();
+    };
+
+    dataviewService.updateDataviewItem = function updateDataviewItem (id, item) {
+      dataviewService.dataview.updateItem(id, item);
+    };
+
+    dataviewService.hasData = function hasData (){
+      return dataviewService.dataview.getItems().length > 0;
+    };
+
+    dataviewService.setGrouping = function setGrouping (g){
+      dataviewService.grouping = g;
+    };
+
+    dataviewService.getGrouping = function getGrouping (){
+      return dataviewService.grouping;
+    };
+
+    dataviewService.groupBy = function groupBy(targetGroup) {
+
+      dataviewService.setGrouping(targetGroup);
+
+      function groupByTransaction() {
+        dataviewService.dataview.setGrouping({
+          getter: 'trans_id',
+          formatter: formatTransactionGroup,
+          comparer : function (a, b) {
+            var x =  parseFloat(a.groupingKey.substr(3));
+            var y =  parseFloat(b.groupingKey.substr(3));
+            return x > y ? 1 : -1;
+          },
+          aggregators: [
+            new Slick.Data.Aggregators.Sum('debit'),
+            new Slick.Data.Aggregators.Sum('credit'),
+            new Slick.Data.Aggregators.Sum('debit_equiv'),
+            new Slick.Data.Aggregators.Sum('credit_equiv')
+          ],
+          aggregateCollapsed: dataviewService.aggregates,
+          lazyTotalsCalculation : true
+        });
+      }
+
+      function groupByAccount () {
+        dataviewService.dataview.setGrouping({
+          getter: 'account_id',
+          formatter: function(g) {
+            var account_txt;
+            if (dataviewService.dataLoaderService.getAccountStore() && g.rows[0].account_number) {account_txt = dataviewService.dataLoaderService.getAccountStore().get(g.rows[0].account_number).account_txt || '';}            
```

Comment:
Man, this is a long line.  Any way you can cache some variables to make it not quite so hard to read?  I'm thinking `var accounts = dataviewService.dataLoaderService.getAccountStore()` would help, and maybe doing a bit more than that.